internal/integration/provision/upgrade.go

```diff
@@ -367,6 +362,11 @@ func (suite *UpgradeSuite) waitForClusterHealth() {
 
 // runE2E runs e2e test on the cluster.
 func (suite *UpgradeSuite) runE2E(k8sVersion string) {
+	if suite.spec.WorkerNodes == 0 {
+		// no worker nodes, should make masters schedulable
+		suite.untaint("master-1")
```

Comment:
for single node upgrade test, we need to untaint the master, otherwise sonobuoy doesn't have a place to run